example/z-docker-opam/dune-project

```diff
@@ -0,0 +1,15 @@
+(lang dune 2.0)
```

Comment:
Is this file necessary for an example, given that it is a second set of redundant boilerplate?

Reply:
I'm not sure I understand: all of the examples include a `dune-project` as they are self-contained. What did you have in mind?

Reply:
This `dune-project` is used to generate an `opam` file, which is then committed. The information in the two files is redundant. All the other examples have only `(lang dune 2.0)` in their `dune-project` files.

Reply:
Isn't it because the other examples don't have an opam file?
I think generating the opam file with `dune-project` is the recommended workflow, especially for beginners, and the line `This file is generated by dune, edit dune-project instead` at the top of the opam file makes it clear that this file should not be edited.

Reply:
I'm not aware of it being recommended, and it seems like more to learn for a benefit that is unclear at this point. It's an extra layer. I also don't use it myself, and no, it's not because they are *meant* to be used with esy (they are not, they build with opam as well, you just use the switch's `dream`). See this early state of the examples: https://github.com/aantron/dream/tree/65d060b9b0da77150f4f67d5252a45568226efc1/example/1-hello.

Reply:
I think having everything in `dune-project` does make it simpler for beginners: 

> it seems like more to learn

On the opposite, it removes one thing to learn, developers can just put all of their configuration in `dune-project` and they "only" have to learn one syntax instead of two.

But I don't have any strong opinion, I'll remove the `package` stanza from the `dune-project` 🙂

Reply:
> On the opposite, it removes one thing to learn, developers can just put all of their configuration in `dune-project` and they "only" have to learn one syntax instead of two.

No, it's a third syntax, since you now need to understand all the extra things for that stanza, which are complex and unique to that stanza, yet the decisive file is still `opam`, lint still works on that, etc. If the `opam` client starts parsing `dune-project` files, or gets replaced by a client that does, then I will agree with this view :)

Reply:
Done 🙂